opentelemetry-sdk/src/logs/log_processor.rs

```diff
@@ -263,7 +263,94 @@ type LogsData = Box<(LogRecord, InstrumentationScope)>;
 /// let provider = LoggerProvider::builder()
 ///     .with_log_processor(processor)
 ///     .build();
+/// ```
+///
+/// **Memory Management in BatchLogProcessor**
+///
+/// The `BatchLogProcessor` manages memory through the following stages of log processing:
+///
+/// 1. **Record Ingestion**:
+///    - Each `LogRecord` is **cloned** upon entering the processor.
+///    - `LogRecordAttributes` utilize a hybrid memory model:
+///      - First 5 attributes are **stack-allocated**.
+///      - Adding additional attributes trigger **heap allocation** in a dynamically growing vector.
+///    - The `LogRecord` and its associated `InstrumentationScope` are **boxed together**
+///      to allocate them on the heap before entering the queue. This means:
+///      - The `LogRecord`'s inline attributes (if any) are moved to the heap as part of the boxed structure.
+///      - Any dynamically allocated data already on the heap (e.g., strings, overflow attributes) remains unaffected.
+///      - Ownership of the boxed data is transferred to the queue, ensuring it can be processed independently of the original objects.
+///
+/// 2. **Queue Management**:
+///    - Uses **two bounded synchronous channels** (`sync_channel`):
+///      - One for **log records** (`logs_sender` and `logs_receiver`).
+///      - Another for **control messages** (`message_sender` and `message_receiver`).
+///
+///    - **Log Record Queue**:
+///      - Stores log records as **heap-allocated** `Box<(LogRecord, InstrumentationScope)>`.
+///      - The queue size is configurable and defined by `max_queue_size`.
+///      - If the queue is full:
+///        - New log records are **dropped**, and a warning is logged the first time this happens.
+///        - Dropped records are counted for reporting during shutdown.
+///
+///    - **Control Message Queue**:
+///      - Stores control messages (`BatchMessage`) to manage operations like exporting, force flushing, setting resources, and shutting down.
+///      - The control message queue has a fixed size (e.g., 64 messages).
+///      - Control messages are processed with higher priority, ensuring operational commands are handled promptly.
```

Comment:
is there anything we do to process them with higher priority? This does not match the implementation.